### PR TITLE
Move code from yii-cycle

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Comment by arogachev** [comment]
Need to fix build for PHP 8.3 and Windows combination.

**Review by samdark — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -26,55 +28,18 @@ The package ...
...
-./vendor/bin/rector
-```
-
-### Dependencies
-
-Use [ComposerRequireChecker](https://github.com/maglnet/ComposerRequireChecker) to detect transitive 
-[Composer](https://getcomposer.org/) dependencies.
+- [English](docs/guide/en)
+- [Russian](docs/guide/ru)
+- [Spanish](docs/guide/es)
```
Comment (vjik) [inline README.md:38]:
```suggestion
- Guide: [English](docs/guide/en), [Russian](docs/guide/ru), [Spanish](docs/guide/es)
```

File: README.md
Comment (vjik) [inline README.md:1]: Add link to internals

File: composer.json [hunk 2]
```diff
@@ -1,19 +1,23 @@
 {
-    "name": "yiisoft/_____",
+    "name": "yiisoft/data-cycle",
     "type": "library",
-    "description": "_____",
+    "description": "Cycle ORM query adapter for yiisoft/data data providers",
```
Comment (vjik) [inline composer.json:4]:
```suggestion
    "description": "Cycle ORM query adapter for yiisoft/data",
```

Package contain not only data providers, writer also.

File: docs/guide/en/entity-reader.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  uses it.
+
+## Examples
+
+Let's implement a repository to work with articles table. We want a method to get public articles `findPublic()` but
+it would not return ready articles collection or select query. Instead, it will return `EntityReader`:
+
+```php
+use Yiisoft\Data\Reader\DataReaderInterface;
+use \Yiisoft\Yii\Cycle\Data\Reader\EntityReader;
```
Comment (vjik) [inline docs/guide/en/entity-reader.md:31]: Need change namespace in docs

**Review by vjik — CHANGES_REQUESTED** [review]

**Pull request merged**